csfiedor

csfiedor opened the pull request:
pull request for code review

csfiedor committed:
Assignment 1 upload

Copying from public repo.

csfiedor committed:
update README

csfiedor committed:
Update README

csfiedor committed:
update README

csfiedor committed:
fix README typos

gmoore10 commented on app/public/stylesheets/style.css:
```diff
@@ -0,0 +1,80 @@
+  clear: both;
+}
+
+#form > input {
```
I would recommend combining the CSS for all of the direct descendants of the #form element.

Something like this would make your CSS file a bit smaller:

`#form > select, #form > button, #form > input {
    float: left;
    clear: both;
    height: 35px;
    width: 100%;
    border-radius: 20px;
    margin-bottom: 5px;
    padding-left: 10px;
}`

This would allow you to combine many of the css properties that are the same between multiple css selectors.

gmoore10 commented on app/routes/index.js:
```diff
@@ -0,0 +1,22 @@
+var router = express.Router();
+var formBuilder = require('../tools/formTool.js');
+
+var input1 = {field: 'input', type: 'text', id: 'firstName', label: 'First Name'};
```
To make this project easier to use, I would suggest placing these objects into a JSON file and importing and parsing them. This will allow other developers to re-use the JSON file and output the desired elements automatically.

gmoore10 commented on app/routes/index.js:
```diff
@@ -0,0 +1,22 @@
+
+/* GET home page. */
+router.get('/', function(req, res, next) {
+  var form = formBuilder(input1, input2, select, button);
```
You can then change the formBuilder method to accept the entire JSON array instead of hard coding the inputs into the method signature.

csfiedor committed:
Unit testing & DWA Style Guide

Unit testing for formTool added. Code enhanced with DWA Style Guide/AirBnB Style Guide.

csfiedor committed:
update README

csfiedor committed:
update README

csfiedor committed:
update README

reactivepixel commented on app/package.json:
```diff
@@ -0,0 +1,38 @@
+{
```
The app folder should not be a wrapping folder for your codebase's logic. See the video in the best practices section regarding folder structure that explores this in-depth.

reactivepixel commented on app/routes/index.js:
```diff
@@ -0,0 +1,44 @@
+
+
+/* GET home page. */
+router.get('/', (req, res) => {
```
You are going beyond the minimum assignment requirements (and do not specify this with your submission) by including an express server to build and render the form being created, however, you stop short of meaningfully adding  functionality. It would be nice if you built an interface to allow inputs and display the outputs, both rendered and raw text that would allow for full use of the tool you have developed.

reactivepixel commented on app/tools/formTool.js:
```diff
@@ -0,0 +1,37 @@
+function formBuilder(...args) {
```
Good use of the ...args aspect of ES6 but why define the function according to ES5 standards? It would be better written as an arrow function to have unity with the newer standard.

reactivepixel commented on app/tools/formTool.js:
```diff
@@ -0,0 +1,37 @@
+  let form = '';
+  // check for data objects
+  if (args.length > 0) {
+    let i;
```
```i``` has no semantic value

reactivepixel commented on app/tools/formTool.js:
```diff
@@ -0,0 +1,37 @@
+  // check for data objects
+  if (args.length > 0) {
+    let i;
+    let x;
```
```x``` has no semantic value

reactivepixel commented on app/tools/formTool.js:
```diff
@@ -0,0 +1,37 @@
+    // through data create specified input type
+    for (i = 0; i < args.length; i += 1) {
+      if (args[i].field === 'input') {
+        form += '<label for="' + args[i].id + '" class="label">' + args[i].label + '</label>';
```
Very good use of the ES6 args usage to efficiently handle the form items but what if no label was defined? Should there not be  conditional logic to resolve that? Especially when your documentation makes no mention of explicitly needing the label defined.

reactivepixel commented on app/tools/formTool.js:
```diff
@@ -0,0 +1,37 @@
+        form += '<label for="' + args[i].id + '" class="label">' + args[i].label + '</label>';
+        form += '<' + args[i].field + ' type="' + args[i].type + '" id="' + args[i].id + '"/>';
+      } else if (args[i].field === 'select') {
+        form += '<label for="' + args[i].id + '" class="label">' + args[i].label + '</label>';
```
Duplicate logic: ```line 10```

reactivepixel commented on app/tools/formTool.js:
```diff
@@ -0,0 +1,37 @@
+        form += '<' + args[i].field + ' id="' + args[i].id + '">';
+        if (args[i].options) {
+          for (x = 0; x < args[i].options.length; x += 1) {
+            form += '<option value="' + args[i].options[x] + '">' + args[i].options[x] + '</option>';
```
Here's a great example of ```i``` and ```x``` having no semantic value. As another developer, looking at this code, I need to spend time deciphering what you have logically constructed.

reactivepixel commented on app/tools/formTool.js:
```diff
@@ -0,0 +1,37 @@
+        form += '<label for="' + args[i].id + '" class="label">' + args[i].label + '</label>';
+        form += '<' + args[i].field + ' id="' + args[i].id + '">';
+        if (args[i].options) {
+          for (x = 0; x < args[i].options.length; x += 1) {
```
Why are there no comments on this file? This is not release ready code.

reactivepixel commented on app/views/layout.pug:
```diff
@@ -0,0 +1,8 @@
+    title= title
+    script(src='http://www.chaijs.com/chai.js' type='text/javascript')
+    link(rel='stylesheet', href='/stylesheets/style.css')
+  body
```
Personal comment: Why do you like jade (pug). Its so.... white space dependent 🤕

reactivepixel commented on app/tools/formTool.js:
```diff
@@ -0,0 +1,37 @@
+        form += '</' + args[i].field + '>';
+      } else if (args[i].field === 'button') {
+        form += '<' + args[i].field + ' type="' + args[i].type + '" id="' + args[i].id + '">' + args[i].label + '</' + args[i].field + '>';
+      } else if (args[i].field === 'label') {
```
Partially flawed logic: You have labels included in other conditionals. I understand the need for a label separate from other instances, but here's another situation where you failed to document this particular situation.

reactivepixel commented on README.md:
```diff
@@ -1 +1,22 @@
+1. clone or download repo
+2. navigate to root of app directory
+3. install dependencies `npm install`
+
```
Documentation: By not providing examples of how to use the tool, what the inputs and outputs of it are you are severely limiting its usage by other devs and yourself, in several months.
